server/src/routes/attainable.ts

```diff
@@ -0,0 +1,140 @@
+// SPDX-FileCopyrightText: 2023 The Aalto Grades Developers
+//
+// SPDX-License-Identifier: MIT
+
+import express, { Router } from 'express';
+
+import { addAttainable, updateAttainable } from '../controllers/attainable';
+import { handleInvalidRequestJson } from '../middleware';
+import { controllerDispatcher } from '../middleware/errorHandler';
+
+export const router: Router = Router();
+
+/**
+ * @swagger
+ * definitions:
+ *   AddAttainment:
+ *     type: object
+ *     description: Information for adding a new study attainment.
+ *     properties:
+ *       parentId:
+ *         type: number
+ *         required: false
+ *         description: Optional parent attainment ID to which the study attainment belongs to.
+ *       name:
+ *         type: string
+ *         required: true
+ *         description: Study attainment name.
+ *       executionDate:
+ *         type: string
+ *         required: true
+ *         description: Date when the attainment is completed (e.g. deadline date or exam date).
+ *       expiryDate:
+ *         type: string
+ *         required: true
+ *         description: Date when the attainment expires.
+ *   Attainment:
+ *     description: Study attainment information.
+ *     allOf:
+ *       - type: object
+ *         properties:
+ *           id:
+ *             type: number
+ *             required: true
+ *             description: Study attainment ID.
+ *           courseId:
+ *             type: number
+ *             required: true
+ *             description: Course ID to which the study attainment belongs to.
+ *           courseInstanceId:
+ *             type: number
+ *             required: true
+ *             description: Course instance ID to which the study attainment belongs to.
+ *       - $ref: '#/definitions/AddAttainment'
+ */
+
+/**
+ * @swagger
+ * /v1/courses/{courseId}/instances/{instanceId}/attainments:
+ *   post:
+ *     tags: [Attainment]
+ *     description: Add a new study attainment.
+ *     requestBody:
+ *       description: New study attainment data.
+ *       content:
+ *         application/json:
+ *           schema:
+ *             $ref: '#/definitions/AddAttainment'
+ *     responses:
+ *       200:
+ *         description: New study attainment added succesfully.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: '#/definitions/Attainment'
+ *       400:
+ *         description: Creation failed, due to validation errors or missing parameters.
+ *         content:
+ *           application/json:
+ *             schema:
+ *               $ref: '#/definitions/Failure'
+ *       404:
+ *         description: Course, course instance, or parent study attainment does not exist.
```

Comment:
I don't think `404` is appropriate if the parent attainment doesn't exist, since it's not included in the URL.

I think I already mentioned this in a previous review, but we've used `422` in `POST /v1/courses/:courseId/instances` when the ID of the given teacher in charge is invalid.

https://developer.mozilla.org/en-US/docs/web/http/status/422

> The HyperText Transfer Protocol (HTTP) 422 Unprocessable Content response status code indicates that the server understands the content type of the request entity, and the syntax of the request entity is correct, but it was unable to process the contained instructions.